### ISSUE TITLE
VOC on yolov2

### ISSUE THREAD

**Issue by xiao1228** [andy-yun/pytorch-0.4-yolov3#49]
Hi,

Thank you for the code!
Without changing the code for yolov2. And using below parameters. The highest mAP I got is at 90 epoch which is only 0.7222 on VOC. I am not sure if I have missed anything?
anchors = 1.1468, 1.5021, 2.7780, 3.4751, 4.3845, 7.0162, 8.2523, 4.2100, 9.7340, 8.682
object_scale=3
noobject_scale=1
class_scale=1
coord_scale=1

Also, for the training and testing, is image size only 416*416 or is it trained and tested with multiscale? 

Thank you for the help!

**Comment by andy-yun** [comment]
I think there is no missed point. it is normal. 
the results are varied with other conditions such as random number, 
Data augmentation etc. Therefore, you should try multiple times.
When you train epoch iteration, the model with 90 epochs may not show the best results.
You should check the intermittent results and select the best model which shows high correct percent.

You can check image.py. In training mode, the image width and height are varied with various conditions.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by xiao1228** [comment]
Hi Andy,
Thank you for the response.
When you mean try multiple times, what parameters are you changing when you try?

Also I found the code for training with different size. Thank you.

**Comment by andy-yun** [comment]
I mean you should train multiple times for the same data. Or you use -l options to find the best trained model.

**Comment by xiao1228** [comment]
ok.. thank you!..however, still cannot get 74..

**Comment by andy-yun** [comment]
um.. try the following conditions:
batch=16
subdivisions=16


I attached my yolo-voc.cfg.
[yolo-voc.zip](https://github.com/andy-yun/pytorch-0.4-yolov3/files/2879564/yolo-voc.zip)

**Comment by andy-yun** [comment]
I once more trained the voc using yolov2. The results are follows:
after the 100th iterated training process, I checked the performance by 100th epoched model, localmax model.
2019-02-19 20:50:11 [010] correct: 7999, precision: 0.553258, recall: 0.664921, fscore: 0.603967
2019-02-19 22:11:07 [020] correct: 8282, precision: 0.535013, recall: 0.688446, fscore: 0.602103
2019-02-19 23:36:01 [030] correct: 8592, precision: 0.596957, recall: 0.714214, fscore: 0.650338
2019-02-20 01:01:46 [040] correct: 8845, precision: 0.571014, recall: 0.735245, fscore: 0.642800
2019-02-20 02:28:03 [050] correct: 8954, precision: 0.611821, recall: 0.744306, fscore: 0.671587
2019-02-20 03:54:14 [060] correct: 8888, precision: 0.635674, recall: 0.738820, fscore: 0.683372
2019-02-20 05:20:04 [070] correct: 8880, precision: 0.628139, recall: 0.738155, fscore: 0.678713
2019-02-20 06:46:27 [080] correct: 8901, precision: 0.627273, recall: 0.739900, fscore: 0.678942
2019-02-20 08:12:44 [090] correct: 8909, precision: 0.627394, recall: 0.740565, fscore: 0.679294
2019-02-20 09:39:16 [100] correct: 8886, precision: 0.630257, recall: 0.738653, fscore: 0.680159

When 100th iterated train model is used,
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           Results:
------------------------------
aeroplane       0.751
bicycle         0.801
bird            0.755
boat            0.611
bottle          0.482
bus             0.823
car             0.826
cat             0.902
chair           0.533
cow             0.788
diningtable     0.748
dog             0.881
horse           0.871
motorbike       0.832
person          0.757
pottedplant     0.477
sheep           0.756
sofa            0.775
train           0.860
tvmonitor       0.745
==============================
 Average        0.749
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When the localmax.weights is used
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           Results:
------------------------------
aeroplane       0.736
bicycle         0.807
bird            0.758
boat            0.583
bottle          0.473
bus             0.818
car             0.822
cat             0.904
chair           0.529
cow             0.787
diningtable     0.745
dog             0.882
horse           0.867
motorbike       0.828
person          0.761
pottedplant     0.476
sheep           0.743
sofa            0.761
train           0.860
tvmonitor       0.731
==============================
 Average        0.743
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

As you compared the results, the localmax always did not show the best performance.
